elide-core/src/main/java/com/yahoo/elide/core/utils/TypeHelper.java

```diff
@@ -156,4 +158,23 @@ public static String getFieldAlias(String tableAlias, String fieldName) {
                         .map(ClassType::of)
                         .collect(Collectors.toSet());
     }
+
+    public static <T> T merge(T original, T update) {
+        final ClassType<?> origClz = ClassType.of(original.getClass());
+        Object returnObject = null;
+        try {
+            returnObject = origClz.newInstance();
+            final Field[] fields = origClz.getDeclaredFields();
+            for (Field field : fields) {
+                field.setAccessible(true);
+                final Object origValue = field.get(original);
+                final Object updateValue = field.get(update);
+                Object value = (updateValue != null && updateValue != origValue) ? updateValue : origValue;
```

Comment:
Thought more about this approach.  I think the issue here is what if the client actually set null in the provided attribute field (what if they wanted to null out a field).  I don't think we can do the merge on the actual objects (since we lose important information about which fields the client actually sent along in the request).

Reply:
I have tried your recommendation and faced several issues. Probably, I do not understand your recommendation entirely. Lets clarify one thing I am stuck with:

What is exactly `original` parameter in the function `updateComplexAttribute(EntityDictionary dictionary, Map updated, Object original, RequestScope scope)`?  If it is [original value:]
(https://github.com/yahoo/elide/blob/3a19d8767497ce15d377ed7f4991c86a6732cefd/elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java#L424)
then issue happens with this `dictionary.isComplexAttribute(original.class, field)`.

So I need your further advise/support if you want me to follow your recommendation. I pushed the change in case you want to look into it.

Reply:
Sorry for the confusion.  Original is the current value for the attribute being modified.  Maybe we should call it currentValue or something instead.  The link you provided in the above description is correct.  What is the issue that happens when you call dictionary.isComplexAttribute?

Reply:
It is `java.lang.IllegalArgumentException: Unbound Entity ClassType{cls=class XYZ} ` because we have passed in a original value of the complex attribute into the call while it is expecting an elide model.

Reply:
Elide now binds both Elide models and complex attributes in the dictionary.

dictionary.isComplexAttribute should be safe to call on any Elide model or any complex attribute.  What class is it being called on when it barfs?

Reply:
We can chat about it on discord if that is simpler.

Reply:
Provided more info of the stack trace on discord. 
Basically, I test the demo project in #2283 with the change in this PR.
and class `Address` (embeddable class) is being called in this case.

Reply:
Hmm.  I don't see this issue.  I added an Address type into the sample project and the dictionary finds it.  Can you publish a failing test on this branch or in a branch on the example project?

Reply:
Sorry, no such issue! Some how my IDE failed to use latest version of the local maven dependency.